deploifai/api/__init__.py

```diff
@@ -298,6 +298,49 @@ def get_cloud_profiles(self, workspace=None) -> typing.List[CloudProfile]:
             ]
         return cloud_profiles
 
+    def create_cloud_profile(self, provider, name, credentials, workspace, fragment):
+        mutation = """
+            mutation(
+              $whereAccount: AccountWhereUniqueInput!
+              $data: CloudProfileCreateInput!
+            ) {
+              createCloudProfile(whereAccount: $whereAccount, data: $data) {
+                ...cloud_profile
+              }
+            }
+        """
+
+        variables = {
+            "whereAccount": {"username": workspace["username"]},
+            "data": {
+                "name": name,
+                "provider": provider.value,
+            },
+        }
+
+        if provider.value == "AWS":
+            variables["data"]["awsCredentials"] = credentials
+        elif provider.value == "AZURE":
+            variables["data"]["azureCredentials"] = credentials
+        else:
```

Comment:
check `provider.value == "GCP"` as well instead of just else because we may have more cloud providers in the future